### PR TITLE
Feat/adyen ios plugin v1

### PR DESCRIPTION
**Pull request by abhinax4991**

### CODE REVIEW

File: src/ios/Adyen.swift [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+                self.presentComponent(with: session)
+            case let .failure(error):
+                self.presentAlert(with: error)
+            }
+        }
+        
+    }
+    
+    
+    func presentt(viewController: UIViewController, completion: (() -> Void)?) {
```
Comment (tyrantbrian) [inline src/ios/Adyen.swift:58]: do we need to name this. 'presentt' with two ts?

File: src/ios/InitialDataFlowProtocolV1.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    func requestAdyenSessionConfiguration(sessionId: String, sessionData: String, context: AdyenContext, completion: @escaping (Result<AdyenSession.Configuration, Error>) -> Void)
+    
+    func generateContext(clientKey: String, currencyCode: String, countryCode: String, value: Int) -> AdyenContext?
+    
+}
+
+extension InitialDataFlowProtocolV1 {
+    private var environmentMapping: [String: Environment] {
+        return [
+            "AU": .liveAustralia,
```
Comment (tyrantbrian) [inline src/ios/InitialDataFlowProtocolV1.swift:22]: Document this on the read me. Also the read me looks pretty bad, please improve it.

File: src/ios/Adyen.swift [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+        }
+        dropInComponent?.finalizeIfNeeded(with: true) { [weak self] in
+            guard let self else { return }
+            if let presentedViewController = topViewController.presentedViewController {
+                presentedViewController.dismiss(animated: true) { [weak self] in
+                    print("dismissing top view controller")
+                    self?.handlePluginResult(resultCode: "Cancelled" , callbackId: self!.callbackId)
+                }
+            } else {
+                //                    topViewController.present(alertController, animated: true, completion: nil)
```
Comment (tyrantbrian) [inline src/ios/Adyen.swift:206]: Do we just ignore this?

File: src/ios/Adyen.swift [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+            }
+            
+        }
+        
+    }
+    
+    func handlePluginResult(resultCode: String, callbackId: String) {
+        switch resultCode {
+        case "Authorised", "Received", "Pending", "PresentToShopper", "Cancelled":
```
Comment (tyrantbrian) [inline src/ios/Adyen.swift:215]: case "Authorised":
case "Received":
case "Pending":
case "PresentToShopper":
case "Cancelled":
self.sendSuccessResult(resultCode: resultCode, callbackId: callbackId) break

Do they not have breaks?

**Review by tyrantbrian — CHANGES_REQUESTED** [review]

File: src/ios/Adyen.swift (hunk 4 above)
Comment (abhinax4991) [reply src/ios/Adyen.swift:215]: swift does not require breaks.

File: src/ios/Adyen.swift (hunk 3 above)
Comment (abhinax4991) [reply src/ios/Adyen.swift:206]: it was breaking the code. so i have kept it commented

**Pull request merged**